File: Data Structures & Algorithms/two-integer-sum/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        let mp = new Map<number, number>();

        for (let i = 0; i < nums.length; i++) {
            let num = nums[i];
            let diff = target - num;
            // mp.set(i, diff)
            if (mp.has(num)) {
                return [mp.get(num), i];
            }
            mp.set(diff, i)
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int leftPtr = 0;
        int rightPtr = numbers.size() - 1;

        while (leftPtr < rightPtr) {
            int l = numbers[leftPtr];
            int r = numbers[rightPtr];
            int sum = l + r;
            if (sum == target) {
                return {leftPtr+1, rightPtr+1};
            }
            if (sum < target) leftPtr++;
            else rightPtr--;
        }
        return {};
        
    }
};
